src/commands/register.ts

```diff
@@ -360,4 +364,37 @@ export default class ComponentRegister extends BaseCommand {
   public static getTagFromFlags(flags: any): string {
     return flags.environment ? `${ENV_TAG_PREFIX}${flags.environment}` : flags.tag;
   }
+
+  private async generateDependenciesWarnings(component_dependencies: Dictionary<string>, account_id: string) {
+    const existing_account_components: Dictionary<string> = {};
+    const { rows, _ } = (await this.app.api.get(`/accounts/${account_id}/components`)).data;
```

Comment:
This endpoint will return the most recently registered component version for up to 100 different components, so I don't think that this covers all of the bases. The version this function is looking for could be two versions ago, or component number 101 registered to the account. I don't think that we really want to overload the logic to find the latest component versions, so we might want to add another endpoint to check existing component versions given a set of component version names. We could call a different endpoint once per dependency to check if it exists, but that's not a great option. @mueschm what are your thoughts here?

Reply:
Your update does not check that the correct tag exists though and just that the component exists. So if a user wants V1 that may not exist but latest would.

I do like the idea of a dependency endpoint. We can pass up the list of component names and versions and it just returns if they exist on the account or not. That would solve the problem, be efficient and provide a possible abstraction that other systems could use if needed.

Reply:
Sounds good. I'll make the changes.